packages/interfaces/src/relay.ts

```diff
@@ -8,11 +8,17 @@ import type {
 } from "./message.js";
 import type { Callback, SendResult } from "./protocols.js";
 
+export type Observer<T extends IDecodedMessage> = {
+  decoder: IDecoder<T>;
+  callback: Callback<T>;
+};
+
 export interface IRelay extends GossipSub {
   send: (encoder: IEncoder, message: IMessage) => Promise<SendResult>;
   addObserver: <T extends IDecodedMessage>(
     decoder: IDecoder<T>,
     callback: Callback<T>
   ) => () => void;
+  observers: Map<string, Set<Observer<any>>>;
```

Comment:
I think it is not correct to use `any` here. I would try to this generic like
```ts
type Observers<T> = Map<string, Set<Observer<T extends IDecodedMessage>>>;
```
But I assume it is not feasible since it can be Observer over messages with different data structures. 
In that case `unknown` is a better option because it will prevent any usage of value without explicit type casting, [here](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type)


@fryorcraken, why `observers` are public? I though of it as more internal API.

Reply:
https://github.com/waku-org/js-waku/issues/1142

Reply:
observers are required for proper logging, to see which relay channels are currently open.

Reply:
@weboko Users wants to be able to track observers currently registered.

Because the need is only logging purposes, I wonder if there is a better way to expose this by may providing a `dumpObservers` function?

Reply:
@weboko @danisharora099 do you think we can/should make it `unknown` instead of `any` internally too?

Reply:
> Because the need is only logging purposes, I wonder if there is a better way to expose this by may providing a dumpObservers function?

If the only reason is logging then we should figure out a better way indeed
exposing something for that kind of reason can break API in future when someone already be using it

> @weboko @danisharora099 do you think we can/should make it unknown instead of any internally too?

If it is something we don't now at a time of using then we should use `unknown` I think. In case we will need to use it as something defined we should do runtime checking such as `typeof` and typescript type casting. This is a bit painful but it helps to prevent nasty bugs

Reply:
Note that the usage of `unknown` did not seem straightforward so I have done that in a separate PR: https://github.com/waku-org/js-waku/pull/1203

Reply:
I proposed another solution in the issue https://github.com/waku-org/js-waku/issues/1142#issuecomment-1449095241